pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -310,109 +306,92 @@ pub async fn disk_usage_eviction_task_iteration_impl<U: Usage>(
         "running disk usage based eviction due to pressure"
     );
 
-    let mut lru_candidates: Vec<(_, LocalLayerInfoForDiskUsageEviction)> = Vec::new();
-
-    // get a snapshot of the list of tenants
-    let tenants = tenant::mgr::list_tenants()
-        .await
-        .context("get list of tenants")?;
-
-    {
-        let mut tmp = Vec::new();
-        for (tenant_id, _state) in &tenants {
-            let flow = extend_lru_candidates(
-                Mode::RespectTenantMinResidentSize,
-                *tenant_id,
-                &mut lru_candidates,
-                &mut tmp,
-                cancel,
-            )
-            .await;
-
-            if let ControlFlow::Break(()) = flow {
-                return Ok(IterationOutcome::Cancelled);
-            }
-
-            assert!(tmp.is_empty(), "tmp has to be fully drained each iteration");
+    let candidates = match collect_eviction_candidates(cancel).await? {
+        EvictionCandidates::Cancelled => {
+            return Ok(IterationOutcome::Cancelled);
         }
-    }
+        EvictionCandidates::Finished(partitioned) => partitioned,
+    };
 
-    if cancel.is_cancelled() {
-        return Ok(IterationOutcome::Cancelled);
+    // Debug-log the list of candidates
+    let now = SystemTime::now();
+    for (i, (partition, candidate)) in candidates
+        // XXX this clone is costly
+        .clone()
+        .into_iter_in_eviction_order()
+        .enumerate()
+    {
+        debug!(
```

Comment:
I guess I'll have to bite the bullet and implement `iter_in_eviction_order()` as well?

Reply:
Hm, the `MinResidentSizePartition` is effectively `overage`, but more expressive and outside of `EvictionCandidates`.

I'll switch the code back to a single list of `(MinResidentSizePartition, EvictionCandidates)` tuples. 

That eliminates the need for `iter_in_eviction_order` alltogether.